frontend/src/pages/Question.tsx

```diff
@@ -1,11 +1,63 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import SubmissionWidget from "../components/SubmissionWidget";
+import { QuestionsPublic, Question, Document } from "../models/questions";
+import LeftSideBar from "../components/LeftSideBar";
+import Markdown from "react-markdown";
+
+const QuestionPage = () => {
+  const [questions, setQuestions] = useState<Question[] | null>(null);
+  const [selectedQuestion, setSelectedQuestion] = useState<Question | null>(
+    null
+  );
+  const [globalDocs, setGlobalDocs] = useState<Document[]>([]);
+
+  useEffect(() => {
+    fetch("/api/questions", {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${localStorage.getItem("token")}`,
+      },
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP Error with Status Code: ${response.status}`);
+        }
+        return response.json();
+      })
+      .then((responseData: QuestionsPublic) => {
+        setQuestions(responseData.questions);
+        setGlobalDocs(responseData.global_docs);
+        setSelectedQuestion(responseData.questions[0]);
+      })
+      .catch((error) => {
+        console.error("Error:", error);
+      });
+  }, []);
+
+  // State stores the selected question.
+  const handleQuestionClick = (questionNum: number) => {
+    const question = questions?.find((q) => q.num - 1 === questionNum) || null;
+    setSelectedQuestion(question);
+  };
 
-const Question = () => {
   return (
-    <div className="flex flex-col items-center min-h-screen pt-12 bg-[#fef7ff] text-[#000000] font-sans">
-      THIS IS THE QUESTION PAGE
+    <div className="flex">
+      <LeftSideBar
+        num={questions?.length}
+        onTabClick={handleQuestionClick}
+      ></LeftSideBar>
+      <section className="w-5/6 prose px-3 pt-3">
+        <p className="text-2xl">{`Problem ` + selectedQuestion?.num}</p>
+        <Markdown>{selectedQuestion?.writeup}</Markdown>
```

Comment:
This markdown editor is decent, but it does not catch all markdown I tested. The most imporant of the lot is headers:
# h1 Heading 8-)

## h2 Heading

### h3 Heading

#### h4 Heading

##### h5 Heading

###### h6 Heading

Reply:
Fixed it. Tailwind removes the styling from ReactMarkdown so I made a class that excludes this from happening to the markdown components